src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java

```diff
@@ -233,8 +233,100 @@ void GIVEN_jobs_deployment_WHEN_deployment_finishes_THEN_status_is_uploaded_to_c
+        PublishRequest pr = prs.get(prs.size() - 1);
+        FleetStatusDetails fleetStatusDetails = OBJECT_MAPPER.readValue(pr.getPayload(),
+                FleetStatusDetails.class);
+        assertEquals("ThingName", fleetStatusDetails.getThing());
+        assertEquals(OverallStatus.HEALTHY, fleetStatusDetails.getOverallStatus());
+        assertNotNull(fleetStatusDetails.getComponentStatusDetails());
+        assertEquals(componentNamesToCheck.size(), fleetStatusDetails.getComponentStatusDetails().size());
+        fleetStatusDetails.getComponentStatusDetails().forEach(componentStatusDetails -> {
+            componentNamesToCheck.remove(componentStatusDetails.getComponentName());
+            if (componentStatusDetails.getComponentName().equals("SimpleApp")) {
+                assertEquals("1.0.0", componentStatusDetails.getVersion());
+                assertEquals(State.FINISHED, componentStatusDetails.getState());
+                assertTrue(componentStatusDetails.isRoot());
+            } else {
+                assertFalse(componentStatusDetails.isRoot());
+            }
+        });
+        assertEquals(0, componentNamesToCheck.size());
+        Slf4jLogAdapter.removeGlobalListener(logListener);
+    }
+
+    @Test
+    void GIVEN_deployment_with_no_component_changes_WHEN_deployment_finishes_THEN_status_is_uploaded_to_cloud() throws Exception {
+        CountDownLatch fssPublishLatch = new CountDownLatch(2);
+        logListener = eslm -> {
+            if (eslm.getEventType() != null
+                    && eslm.getEventType().equals("fss-status-update-published")
+                    && eslm.getMessage().contains("Status update published to FSS")) {
+                fssPublishLatch.countDown();
+            }
+        };
+        Slf4jLogAdapter.addGlobalListener(logListener);
+
+        // First deployment adding SimpleApp v1
+        Map<String, String> componentsToMerge = new HashMap<>();
+        componentsToMerge.put("SimpleApp", "1.0.0");
+        LocalOverrideRequest firstRequest = LocalOverrideRequest.builder().requestId("SimpleApp1-1")
+                .componentsToMerge(componentsToMerge)
+                .requestTimestamp(System.currentTimeMillis())
+                .build();
+        submitLocalDocument(firstRequest);
+
+        // Second deployment adding SimpleApp v1 again
+        LocalOverrideRequest secondRequest = LocalOverrideRequest.builder().requestId("SimpleApp1-2")
+                .componentsToMerge(componentsToMerge)
+                .requestTimestamp(System.currentTimeMillis())
+                .build();
+        submitLocalDocument(secondRequest);
+
+        assertTrue(fssPublishLatch.await(180, TimeUnit.SECONDS));
+        verify(mqttClient, atLeastOnce()).publish(captor.capture());
+        List<PublishRequest> prs = captor.getAllValues();
+
+        // Get the last FSS publish request which should have no new component info, since no component statuses have
+        // changed
```

Comment:
if the component had finished, would this have published a change/restarted the component? I'm assuming not - is this correct?

Reply:
No. Shouldn't change the pre-existing behavior in that way.